plugins/woocommerce/src/Admin/API/Init.php

```diff
@@ -54,115 +55,167 @@ public function __construct() {
 	}
 
 	/**
-	 * Init REST API.
+	 * Initialize the API namespaces under WooCommerce Admin.
+	 *
+	 * @return void
 	 */
 	public function rest_api_init() {
-		$controllers           = array();
-		$analytics_controllers = array();
-
 		if ( wc_rest_should_load_namespace( 'wc-admin' ) ) {
-			// Controllers in the wc-admin namespace.
-			$controllers = array(
-				'Automattic\WooCommerce\Admin\API\Notice',
-				'Automattic\WooCommerce\Admin\API\Features',
-				'Automattic\WooCommerce\Admin\API\Experiments',
-				'Automattic\WooCommerce\Admin\API\Marketing',
-				'Automattic\WooCommerce\Admin\API\MarketingOverview',
-				'Automattic\WooCommerce\Admin\API\MarketingRecommendations',
-				'Automattic\WooCommerce\Admin\API\MarketingChannels',
-				'Automattic\WooCommerce\Admin\API\MarketingCampaigns',
-				'Automattic\WooCommerce\Admin\API\MarketingCampaignTypes',
-				'Automattic\WooCommerce\Admin\API\Options',
-				'Automattic\WooCommerce\Admin\API\Settings',
-				'Automattic\WooCommerce\Admin\API\PaymentGatewaySuggestions',
-				'Automattic\WooCommerce\Admin\API\Themes',
-				'Automattic\WooCommerce\Admin\API\Plugins',
-				'Automattic\WooCommerce\Admin\API\OnboardingFreeExtensions',
-				'Automattic\WooCommerce\Admin\API\OnboardingProductTypes',
-				'Automattic\WooCommerce\Admin\API\OnboardingProfile',
-				'Automattic\WooCommerce\Admin\API\OnboardingTasks',
-				'Automattic\WooCommerce\Admin\API\OnboardingThemes',
-				'Automattic\WooCommerce\Admin\API\OnboardingPlugins',
-				'Automattic\WooCommerce\Admin\API\OnboardingProducts',
-				'Automattic\WooCommerce\Admin\API\MobileAppMagicLink',
-				'Automattic\WooCommerce\Admin\API\ShippingPartnerSuggestions',
-			);
+			$this->rest_api_init_wc_admin();
```

Comment:
I'm not sure if I am misunderstanding the diff here, but wouldn't this need to use `lazy_load_namespace` in order to lazy load? `rest_api_init_wc_admin` registers the routes right away.

Reply:
If this is handled via `wc_rest_should_load_namespace` I think I'd want to understand when wc_rest_should_load_namespace/lazy_load_namespace is appropriate.

Reply:
The goal on this diff changed and I didn't go back and update the content of the PR properly.  Initially I planned to lazy load both wc-admin and wc-analytics.  However, I was concerned about the number of extensions that may extend wc-admin compared to wc-analytics and decided to just start with wc-analytics for 10.3.